src/sentry/uptime/detectors/tasks.py

```diff
@@ -0,0 +1,169 @@
+    queue="uptime",
+    time_limit=60,
+    soft_time_limit=55,
+)
+def schedule_detections():
+    """
+    Runs regularly and fires off a task for each detection bucket that needs to be run since
+    this task last ran.
+    """
+    lock = locks.get(
+        SCHEDULER_LOCK_KEY,
+        duration=60,
+        name="uptime.detection.schedule_detections",
+    )
+    try:
+        with lock.acquire():
+            cluster = _get_cluster()
+            last_processed = cluster.get(LAST_PROCESSED_KEY)
+            if last_processed is None:
+                last_processed = timezone.now().replace(second=0, microsecond=0)
+            else:
+                last_processed = datetime.datetime.fromtimestamp(
+                    int(last_processed), tz=datetime.UTC
+                )
+
+            minutes_since_last_processed = int(
+                (timezone.now() - last_processed) / timedelta(minutes=1)
+            )
+            for _ in range(minutes_since_last_processed):
+                last_processed = last_processed + timedelta(minutes=1)
+                process_detection_bucket.delay(last_processed)
+
+            cluster.set(LAST_PROCESSED_KEY, int(last_processed.timestamp()), timedelta(hours=1))
+    except UnableToAcquireLock:
+        # If we can't acquire the lock it just means another task is already handling scheduling,
+        # so just exit
+        metrics.incr("uptime.detectors.scheduler.unable_to_acquire_lock")
+
+
+@instrumented_task(
+    name="sentry.uptime.detectors.tasks.process_detection_bucket",
+    queue="uptime",
+)
+def process_detection_bucket(bucket: datetime.datetime):
+    """
+    Schedules url detection for all projects in this time bucket that saw promising urls.
+    """
+    for project_id, count in get_project_bucket(bucket).items():
+        process_project_url_ranking.delay(project_id, count)
+    delete_project_bucket(bucket)
+
+
+@instrumented_task(
+    name="sentry.uptime.detectors.tasks.process_project_url_ranking",
+    queue="uptime",
+)
+def process_project_url_ranking(project_id: int, project_url_count: int):
+    """
+    Looks at candidate urls for a project and determines whether we should start monitoring them
+    """
+    project = Project.objects.get_from_cache(id=project_id)
+    if not should_detect_for_project(project):
+        return
+
+    for url, url_count in get_candidate_urls_for_project(project)[:5]:
+        if check_url(project, project_url_count, url, url_count):
+            # TODO: On success, we want to mark this project as not needing to be checked for a while
+            break
+    else:
+        # TODO: If we don't find any urls to monitor, we want to increment a counter in redis and check the value.
+        # After a number of failures, we want to stop checking for this project for a while.
+        pass
+
+    delete_candidate_urls_for_project(project)
+
+
+def check_url(project: Project, project_url_count: int, url: str, url_count: int) -> bool:
```

Comment:
`check_url` kind of makes it sound like it's going to make HTTP requests. Maybe a better name could be `url_meets_criteria` or something?

Reply:
maybe it just needs a doc string